service/frontend/workflow_handler.go

```diff
@@ -3167,21 +3177,32 @@ func (wh *WorkflowHandler) UpdateSchedule(ctx context.Context, request *workflow
 		return nil, err
 	}
 
-	input := &schedspb.FullUpdateRequest{
+	executionInfo, startScheduledArgs, err := wh.getScheduleExecutionInfoAndSchedulingArgs(ctx, workflowID, namespaceName.String(), namespaceID)
+	if err != nil {
+		return nil, err
+	}
+	startScheduledArgs.Schedule = request.Schedule
```

Comment:
preserve `Info`. and set `UpdateTime` of `Info` to the current time. basically, do what `processUpdate` in the workflow would do